test/integration/edition_edit_test.rb

```diff
@@ -108,54 +83,170 @@ class EditionEditTest < IntegrationTest
         assert page.has_text?(/English/)
       end
     end
+  end
 
-    context "unpublish tab" do
+  context "unpublish tab" do
+    context "do not have required permissions" do
       setup do
-        click_link("Unpublish")
+        login_as(FactoryBot.create(:user, name: "Stub User"))
+        visit_edition_in_draft
       end
 
-      should "show 'Unpublish' header and 'Continue' button" do
-        within :css, ".gem-c-heading" do
-          assert page.has_text?("Unpublish")
-        end
-        assert page.has_button?("Continue")
+      should "not show unpublish tab when user is not govuk editor" do
+        assert page.has_no_text?("Unpublish")
       end
+    end
 
-      should "show 'cannot be undone' banner" do
-        assert page.has_text?("If you unpublish a page from GOV.UK it cannot be undone.")
+    context "has required permissions" do
+      setup do
+        login_as(FactoryBot.create(:user, :govuk_editor, name: "Stub User"))
+        visit_edition_in_draft
       end
 
-      should "show 'Redirect to URL' text, input box and example text" do
-        assert page.has_text?("Redirect to URL")
-        assert page.has_text?("For example: https://www.gov.uk/redirect-to-replacement-page")
-        assert page.has_css?(".govuk-input", count: 1)
+      context "when state is publish" do
+        setup do
+          visit_edition_in_published
+          click_link("Unpublish")
+        end
+
+        should "show 'Unpublish' header and 'Continue' button" do
+          within :css, ".gem-c-heading" do
+            assert page.has_text?("Unpublish")
+          end
+          assert page.has_button?("Continue")
+        end
+
+        should "show 'cannot be undone' banner" do
+          assert page.has_text?("If you unpublish a page from GOV.UK it cannot be undone.")
+        end
+
+        should "show 'Redirect to URL' text, input box and example text" do
+          assert page.has_text?("Redirect to URL")
+          assert page.has_text?("For example: https://www.gov.uk/redirect-to-replacement-page")
+          assert page.has_css?(".govuk-input", count: 1)
+        end
+
+        should "navigate to 'confirm-unpublish' page when clicked on 'Continue' button" do
```

Comment:
Could do with a minor rewrite for grammar:
```suggestion
        should "navigate to 'confirm-unpublish' page when the 'Continue' button is clicked" do
```